### PR TITLE
kconfig: decouple realmode boot from CONFIG_JAIHOUSE

### PR DESCRIPTION
**Pull request by ZideChen0**
Add CONFIG_REALMODE item so that it's possible to configure other
x86 boards to boot from real mode.

Signed-off-by: Zide Chen <zide.chen@intel.com>

### CODE REVIEW

File: boards/x86/x86_jailhouse/Kconfig.defconfig [hunk 1]
```diff
@@ -8,6 +8,10 @@ config JAILHOUSE
 	bool "Zephyr port to boot as a (x86) Jailhouse inmate cell payload"
 	default y
 
+config REALMODE
```
Comment (nashif) [inline boards/x86/x86_jailhouse/Kconfig.defconfig]: you need to define REALMODE outside of the board to make it available for others, in arch/x86/Kconfig for example, then just set it in the _defconfig of the jailhouse board.

**Review by nashif — CHANGES_REQUESTED** [review]

File: boards/x86/x86_jailhouse/Kconfig.defconfig (hunk 1 above)
Comment (ZideChen0) [reply boards/x86/x86_jailhouse/Kconfig.defconfig]: I see, thank you very much for pointing this out.

Comment (ZideChen0) [reply boards/x86/x86_jailhouse/Kconfig.defconfig]: It seems arch/x86/Kconfig is for hardware feature configurations. Can I put this item in arch/x86/core/Kconfig?

Comment (andrewboie) [reply boards/x86/x86_jailhouse/Kconfig.defconfig]: fine with me

Comment (ZideChen0) [reply boards/x86/x86_jailhouse/Kconfig.defconfig]: Sorry for posting comments before carefully look at the code. 

It seems making more sense to put it in the "Boot Options" in misc/Kconfig, and make it depend on X86. Initially I ignore this file and just thought real mode is for X86 only and should be under arch/x86.

File: arch/x86/core/crt0.S [hunk 2]
```diff
@@ -42,7 +42,7 @@
 	GTEXT(_sys_soc_resume_from_deep_sleep)
 #endif
 
-#ifdef CONFIG_JAILHOUSE
+#ifdef CONFIG_REALMODE
 
 #define JAILHOUSE_INMATE_CS32   0x8
 #define JAILHOUSE_INMATE_DS32   0x18
```
[the remark this answers is not in the document]
Comment (ZideChen0) [reply arch/x86/core/crt0.S]: Yes, will do

**Review by nashif — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request merged**